app/models/commit.rb

```diff
@@ -83,6 +83,19 @@ def refresh_statuses!
     end
   end
 
+  def add_status(status_attributes)
+    previous_status = significant_status
+    statuses.create!(status_attributes)
+    reload
+    new_status = significant_status
+    if (previous_status != new_status) # we only care if something changed
```

Comment:
Hum, you have a very good point here. Maybe the simplest thing to do is to send all the newest statuses for each context. The client can do whatever it want with it.

But if we then need to decide what to do with the hidden and ignored statuses. Should we send them? If so do we give them a special property?